### DISCUSSION TITLE
Can this be run programmatically?

### DISCUSSION THREAD

**Discussion by crypto-kungfu** [v8hid/infinite-zoom-automatic1111-webui#106] [General]
This example lets me create text2img via api but would anyone know how i could run infinitezoom via the same approach?

import json
import base64

import requests

def submit_post(url: str, data: dict):
    """
    Submit a POST request to the given URL with the given data.
    """
    return requests.post(url, data=json.dumps(data))

def save_encoded_image(b64_image: str, output_path: str):
    """
    Save the given image to the given output path.
    """
    with open(output_path, "wb") as image_file:
        image_file.write(base64.b64decode(b64_image))

if __name__ == '__main__':
    txt2img_url = 'http://127.0.0.1:7861/sdapi/v1/txt2img'
    data = {'prompt': 'a dog wearing a hat'}
    response = submit_post(txt2img_url, data)
    save_encoded_image(response.json()['images'][0], 'dog.png')

**Comment by v8hid** [comment]
HI @crypto-kungfu, unfortunately we don't offer API, but if you really need it you can fork the repo and make it happen!
